test/e2e/specs/site-editor/title.spec.js

```diff
@@ -0,0 +1,77 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+test.describe( 'Site editor title', () => {
+	test.beforeAll( async ( { requestUtils } ) => {
+		await requestUtils.activateTheme( 'emptytheme' );
+	} );
+
+	test.afterAll( async ( { requestUtils } ) => {
+		await requestUtils.activateTheme( 'twentytwentyone' );
+	} );
+
+	test( 'displays the selected template name in the title for the index template', async ( {
+		page,
+		pageUtils,
+	} ) => {
+		// Navigate to a template.
+		await pageUtils.visitSiteEditor( {
+			postId: 'emptytheme//index',
+			postType: 'wp_template',
+		} );
+
+		const title = await page.locator(
+			'role=region[name="Header"i] >> role=heading[level=1]'
+		);
+
+		await expect( await title.textContent() ).toEqual(
+			'Editing template: Index'
+		);
```

Comment:
Let's use [`toHaveText()`](https://playwright.dev/docs/test-assertions#locator-assertions-to-have-text)

```suggestion
		await expect( title ).toHaveText(
			'Editing template: Index'
		);
```

and ditto to other places below.

Reply:
Ah, missed that in your previous comment, that's much nicer 👍

Reply:
It also has a nice benefit of being able to auto-retry the assertion until it passes!